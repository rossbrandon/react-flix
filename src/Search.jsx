// @flow

import React, { Component } from 'react';
import ShowCard from './ShowCard';
import Header from './Header';

class Search extends Component {
  state = {
    searchTerm: ''
  };
  props: {
    shows: Array<Show>
  };
  handleSearchTermChange = (event: SyntheticKeyboardEvent & { target: HTMLInputElement }) => {
    this.setState({ searchTerm: event.target.value });
  };
  render() {
    return (
      <div className="search">
        <header>
          <Header searchTerm={this.state.searchTerm} showSearch handleSearchTermChange={this.handleSearchTermChange} />
          <input
            onChange={this.handleSearchTermChange}
            value={this.state.searchTerm}
            type="text"
            placeholder="Search"
          />
        </header>
        <div>
          {this.props.shows
            .filter(
              show =>
                `${show.title} ${show.description}`.toUpperCase().indexOf(this.state.searchTerm.toUpperCase()) >= 0
            )
            .map((show, index) => <ShowCard {...show} key={show.imdbID} id={index} />)}
        </div>
      </div>
    );
  }
}

export default Search;
