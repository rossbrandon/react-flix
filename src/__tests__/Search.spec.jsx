import React from 'react';
import { shallow } from 'enzyme';
import Search from '../Search';
import ShowCard from '../ShowCard';
import preload from '../../data.json';

test('Search renders correctly', () => {
  const component = shallow(<Search />);
  expect(component).toMatchSnapshot();
});

test('Search should render correct amount of shows', () => {
  const component = shallow(<Search shows={preload.shows} />);
  expect(component.find(ShowCard).length).toEqual(preload.shows.length);
});

test('Search should render correct shows after search', () => {
  const searchWord = 'black';
  const component = shallow(<Search shows={preload.shows} />);
  component.find('input').simulate('change', { target: { value: searchWord } });
  const showCount = preload.shows.filter(
    show => `${show.title} ${show.description}`.toUpperCase().indexOf(searchWord.toUpperCase()) >= 0
  ).length;

  expect(component.find(ShowCard).length).toEqual(showCount);
});
