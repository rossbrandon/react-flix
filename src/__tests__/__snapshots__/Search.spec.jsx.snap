// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <header>
    <h1 />
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <ShowCard
      description="Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star."
      id={0}
      imdbID="tt4288182"
      poster="a.jpg"
      title="Atlanta"
      trailer="MpEdJ-mmTlY"
      year="2016-"
    />
    <ShowCard
      description="U.S. Attorney Chuck Rhoades goes after hedge fund king, Bobby \\"Axe\\" Axelrod in a battle between two powerful New York figures."
      id={1}
      imdbID="tt4270492"
      poster="b.jpg"
      title="Billions"
      trailer="_raEUMLL-ZI"
      year="2016–"
    />
    <ShowCard
      description="A television anthology series that shows the dark side of life and technology."
      id={2}
      imdbID="tt2085059"
      poster="bm.jpg"
      title="Black Mirror"
      trailer="jDiYGjp5iFg"
      year="2011–"
    />
    <ShowCard
      description="A high school chemistry teacher diagnosed with inoperable lung cancer turns to manufacturing and selling methamphetamine in order to secure his family's future."
      id={3}
      imdbID="tt0903747"
      poster="bb.jpg"
      title="Breaking Bad"
      trailer="XZ8daibM3AE"
      year="2008–2013"
    />
    <ShowCard
      description="Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years."
      id={4}
      imdbID="tt0944947"
      poster="got.jpg"
      title="Game of Thrones"
      trailer="giYeaKsXnsI"
      year="2011–"
    />
    <ShowCard
      description="A drama centered on the personal and professional lives of five surgical interns and their supervisors."
      id={5}
      imdbID="tt0413573"
      poster="greys.jpg"
      title="Grey's Anatomy"
      trailer="q1pcpgREQ5c"
      year="2005"
    />
    <ShowCard
      description="A bipolar CIA operative becomes convinced a prisoner of war has been turned by al-Qaeda and is planning to carry out a terrorist attack on American soil."
      id={6}
      imdbID="tt1796960"
      poster="h.jpg"
      title="Homeland"
      trailer="KyFmS3wRPCQ"
      year="2011–"
    />
    <ShowCard
      description="An antisocial maverick doctor who specializes in diagnostic medicine does whatever it takes to solve puzzling cases that come his way using his crack team of doctors and his wits."
      id={7}
      imdbID="tt0412142"
      poster="house.jpg"
      title="House M.D."
      trailer="5DIADh4lMq8"
      year="2004–2012"
    />
    <ShowCard
      description="A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him."
      id={8}
      imdbID="tt1856010"
      poster="hoc.jpg"
      title="House of Cards"
      trailer="NTzycsqxYJ0"
      year="2013–"
    />
    <ShowCard
      description="The personal and professional life of Dev, a 30-year-old actor in New York."
      id={9}
      imdbID="tt4635276"
      poster="mon.jpg"
      title="Master of None"
      trailer="6bFvb3WKISk"
      year="2015–"
    />
    <ShowCard
      description="A chronicled look at the criminal exploits of Colombian drug lord Pablo Escobar."
      id={10}
      imdbID="tt2707408"
      poster="n.jpg"
      title="Narcos"
      trailer="U7elNhHwgBU"
      year="2015–"
    />
    <ShowCard
      description="The story of Piper Chapman, a woman in her thirties who is sentenced to fifteen months in prison after being convicted of a decade-old crime of transporting money for her drug-dealing girlfriend."
      id={11}
      imdbID="tt2372162"
      poster="oitnb.jpg"
      title="Orange Is the New Black"
      trailer="th8WT_pxGqg"
      year="2013–"
    />
    <ShowCard
      description="An animated series that follows the exploits of a super scientist and his not-so-bright grandson."
      id={12}
      imdbID="tt2861424"
      poster="rm.jpg"
      title="Rick and Morty"
      trailer="dpoNN-kZ6o8"
      year="2013-"
    />
    <ShowCard
      description="Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper."
      id={13}
      imdbID="tt2575988"
      poster="sv.jpg"
      title="Silicon Valley"
      trailer="69V__a49xtw"
      year="2014–"
    />
    <ShowCard
      description="When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back."
      id={14}
      imdbID="tt4574334"
      poster="st.jpg"
      title="Stranger Things"
      trailer="9Egf5U8xLo8"
      year="2016–"
    />
    <ShowCard
      description="Two Soviet intelligence agents pose as a married couple to spy on the American government."
      id={15}
      imdbID="tt2149175"
      poster="ta.jpg"
      title="The Americans"
      trailer="HjuUkbhsI24"
      year="2013–"
    />
    <ShowCard
      description="Former Senator Selina Meyer finds that being Vice President of the United States is nothing like she hoped and everything that everyone ever warned her about."
      id={16}
      imdbID="tt1759761"
      poster="v.jpg"
      title="Veep"
      trailer="I2HoCG_iyhU"
      year="2012–"
    />
    <ShowCard
      description="Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence."
      id={17}
      imdbID="tt0475784"
      poster="ww.jpg"
      title="Westworld"
      trailer="eX3u0IlBBO4"
      year="2016–"
    />
  </div>
</div>
`;
